plugin/https/proxy.go

```diff
@@ -0,0 +1,145 @@
+package https
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/miekg/dns"
+)
+
+const (
+	dnsMessageMimeType = "application/dns-message"
+
+	// typical Ethernet MTU (1500 bytes) - min IP header size (20 bytes) - UDP header (8 bytes).
+	// It seems like a reasonable limitation for DoH protocol.
+	// However, if you know RFCs that specify this limit, update it.
+	maxDNSMessageSize     = 1472
```

Comment:
why do we care? Let's make the http lib figure this out

Reply:
Default HTTP library doesn't limit http response body. So that in the event of an upstream server compromise, coredns server may be vulnerable to DoS attacks.